overlord/snapshotstate/backend/reader.go

```diff
@@ -309,11 +309,19 @@ func (r *Reader) Restore(ctx context.Context, current snap.Revision, usernames [
 			"--directory", tempdir)
 		cmd.Env = []string{}
 		cmd.Stdin = tr
-		cmd.Stderr = os.Stderr
-		cmd.Stdout = os.Stderr
+		matchCounter := &strutil.MatchCounter{Regexp: matchAll, N: 1}
```

Comment:
Using a match counter that matches everything is a little weird. Perhaps this could use a comment in the code to explain the rationale?

Reply:
with a regexp that matches everything, `strutil.MatchCounter` is a `strutil.LimitedBuffer` with no line length limit, that remembers how many lines it got.

Reply:
I've updated it to make use of the new regexpless match counter